### PR TITLE
Fix for #117 xml_crypto_api.h contains multiple free of unallocated errors

### PR DESCRIPTION
**Pull request by igoropaniuk**
Add additional labels so that cleanup order is correct
Change deallocation order

All test cases are successfully passed

### CODE REVIEW

File: host/xtest/xml/include/xml_crypto_api.h [hunk 1]
```diff
@@ -851,6 +852,7 @@ static TEEC_Result Invoke_Crypto_GetOperationInfoMultiple(
 exit:
```
Comment (jforissier) [inline host/xtest/xml/include/xml_crypto_api.h]: This label is not used anymore, remove it (occurs several times below).

File: host/xtest/xml/include/xml_crypto_api.h [hunk 2]
```diff
@@ -2375,8 +2412,10 @@ static TEEC_Result calculate_digest(
 
 exit:
 	Invoke_Crypto_FreeOperation(c, s, CMD_FreeOperation, &op1);
```
Comment (jforissier) [inline host/xtest/xml/include/xml_crypto_api.h]: This should be done last.

File: host/xtest/xml/include/xml_crypto_api.h [hunk 3]
```diff
@@ -2447,8 +2486,10 @@ static TEEC_Result sign_digest(
...
```
Comment (jforissier) [inline host/xtest/xml/include/xml_crypto_api.h]: Same here.

File: host/xtest/xml/include/xml_crypto_api.h [hunk 4]
```diff
@@ -2511,8 +2552,10 @@ static bool verify_digest(
...
```
Comment (jforissier) [inline host/xtest/xml/include/xml_crypto_api.h]: And here.

File: host/xtest/xml/include/xml_crypto_api.h [hunk 5]
```diff
@@ -2583,8 +2626,10 @@ static TEEC_Result mac_compute_final(
...
```
Comment (jforissier) [inline host/xtest/xml/include/xml_crypto_api.h]: Here also.

File: host/xtest/xml/include/xml_crypto_api.h [hunk 6]
```diff
@@ -2672,8 +2718,10 @@ static TEEC_Result cipher_do_final(
...
```
Comment (jforissier) [inline host/xtest/xml/include/xml_crypto_api.h]: Idem.

File: host/xtest/xml/include/xml_timearithm_api.h [hunk 7]
```diff
@@ -366,45 +367,15 @@ static TEEC_UUID UUID_TTA_testingInternalAPI_Time = {
...
-#define ALLOCATE_SHARED_MEMORY(context, sharedMemory, sharedMemorySize, \
-			       memoryType) \
-	res = AllocateSharedMemory(context, sharedMemory, sharedMemorySize, \
-				   memoryType); \
-	if (res != TEEC_SUCCESS) { \
-		goto exit; \
-	} \
-	memset(sharedMemory->buffer, 0, sharedMemorySize);
-
-#define ALLOCATE_AND_FILL_SHARED_MEMORY(context, sharedMemory, \
```
Comment (jforissier) [inline host/xtest/xml/include/xml_timearithm_api.h]: This one is different, is has 5 parameters. The other versions you have removed (in other .h files) had 6 parameters.
As a result the macro in `xml_common_api.h` does not work in `xml_timearithm_api.h`, and then `xtest_8500.c` fails to compile (it is a GP test).

**Comment by igoropaniuk** [comment]
@d3zd3z Could you please look into this fix? Thanks :)

**Comment by jforissier** [comment]
You need to drop the parentheses in the definition of `ALLOCATE_AND_FILL_SHARED_MEMORY_6`, because one parameter is actually a label. Sorry for the confusing advice above.

```diff
diff --git a/host/xtest/xml/include/xml_common_api.h b/host/xtest/xml/include/xml_common_api.h
index 798253e..fd8a46b 100644
--- a/host/xtest/xml/include/xml_common_api.h
+++ b/host/xtest/xml/include/xml_common_api.h
@@ -40,7 +40,7 @@
                memcpy(sharedMemory->buffer, data, copySize); \
        }
 #define ALLOCATE_AND_FILL_SHARED_MEMORY_6(a,b,c,d,e,f) \
-                       ALLOCATE_AND_FILL_SHARED_MEMORY((a),(b),(c),(d),(c),(e),(f))
+                       ALLOCATE_AND_FILL_SHARED_MEMORY(a,b,c,d,c,e,f)
 
 #define SET_SHARED_MEMORY_OPERATION_PARAMETER(parameterNumber, \
                                              sharedMemoryOffset, \
```
WIth that applied:
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

**Comment by jforissier** [comment]
Code is OK but commit message needs fixing:
- `dbg xml headers`: I don't see the purpose of the `dbg`, can you explain or remove it?
- Typo: Invoke_Crypto_FreeOperationa
- Missing `Signed-off-by:`
- Also no blank line between `Fixes:` and `Tested-by:`, please.

Once the above are taken into account:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by igoropaniuk** [comment]
Tags applied, fixed commit message

**Pull request merged**